controllers/configuration_controller.go

```diff
@@ -1358,3 +1497,61 @@ func GetGitCredentialsSecret(ctx context.Context, k8sClient client.Client, secre
 
 	return secret, nil
 }
+
+// GetTerraformCredentialsSecret will get the secret containing the terraform credentials
+func GetTerraformCredentialsSecret(ctx context.Context, k8sClient client.Client, secretRef *v1.SecretReference) (*v1.Secret, error) {
```

Comment:
Also, these GetFooSecret/GetFooConfigMap can be unified with a generator function.

Reply:
fixed as per review comment.

Reply:
removed GetFooSecret/GetFooConfigMap functions and we call GetSecretOrConfigMap to retrieve the secret/configmap